### PR TITLE
Implement shared queue for OMEZarr

### PR DESCRIPTION
**Pull request by tischi**
Hi @KateMoreva (cc @K-Meech),

I hacked in a `SharedQueue` for OMEZarr. This helps in terms of not creating too many threads when adding very many Sources to BDV.

I would like to merge this as it will enable @constantinpape to have a look at the plate view images.
But it would be good if you could take note of what I did here such that we can make a nice implementation of this in bdv-imageloader.

### CODE REVIEW

File: src/main/java/de/embl/cba/mobie/MoBIE.java [hunk 1]
```diff
@@ -57,7 +58,8 @@
 {
 	public static final int N_THREADS = 8;
 	public static final String PROTOTYPE_DISPLAY_VALUE = "01234567890123456789";
-
+	public static final ExecutorService executorService = Executors.newFixedThreadPool( N_THREADS );
+	public static final SharedQueue sharedQueue = new SharedQueue(Math.max(1, Runtime.getRuntime().availableProcessors()));
```
Comment (tischi) [inline src/main/java/de/embl/cba/mobie/MoBIE.java:62]: @KateMoreva this global queue will be used by all image loaders.

File: src/main/java/de/embl/cba/mobie/n5/zarr/N5OMEZarrImageLoader.java [hunk 2]
```diff
@@ -119,6 +121,12 @@ public N5OMEZarrImageLoader(N5Reader n5Reader, HashMap<String, Integer> axesMap)
 		fetchSequenceDescriptionAndViewRegistrations();
 	}
 
+	public N5OMEZarrImageLoader(N5Reader n5Reader, HashMap<String, Integer> axesMap, BlockingFetchQueues< Callable< ? > > queue ) {
```
Comment (tischi) [inline src/main/java/de/embl/cba/mobie/n5/zarr/N5OMEZarrImageLoader.java:124]: @KateMoreva here is a new constructor that can take this queue.

File: src/main/java/de/embl/cba/mobie/n5/zarr/N5OMEZarrImageLoader.java [hunk 3]
```diff
@@ -291,10 +299,13 @@ private void open() {
 						maxNumLevels = Math.max(maxNumLevels, setupImgLoader.numMipmapLevels());
 					}
 
-					final int numFetcherThreads = Math.max(1, Runtime.getRuntime().availableProcessors());
-					final BlockingFetchQueues<Callable<?>> queue = new BlockingFetchQueues<>(maxNumLevels, numFetcherThreads);
-					fetchers = new FetcherThreads(queue, numFetcherThreads);
-					cache = new VolatileGlobalCellCache(queue);
+					if ( queue == null )
```
Comment (tischi) [inline src/main/java/de/embl/cba/mobie/n5/zarr/N5OMEZarrImageLoader.java:302]: @KateMoreva  we only create a new queue if none is provided

File: src/main/java/de/embl/cba/mobie/n5/zarr/OMEZarrReader.java [hunk 4]
```diff
@@ -38,7 +40,7 @@ private SpimData readFile() throws IOException
         N5OMEZarrImageLoader.logChunkLoading = logChunkLoading;
         N5OmeZarrReader reader = new N5OmeZarrReader(this.filePath, new GsonBuilder());
         HashMap<String, Integer> axesMap = reader.getAxes();
-        N5OMEZarrImageLoader imageLoader = new N5OMEZarrImageLoader(reader, axesMap);
+        N5OMEZarrImageLoader imageLoader = new N5OMEZarrImageLoader(reader, axesMap, MoBIE.sharedQueue);
```
Comment (tischi) [inline src/main/java/de/embl/cba/mobie/n5/zarr/OMEZarrReader.java:43]: @KateMoreva here the global shared queue is injected into the ImageLoader. We should enable this for all ImageLoaders, not only the OMEZarr ones.

Comment (tischi) [reply src/main/java/de/embl/cba/mobie/n5/zarr/OMEZarrReader.java:43]: Good point, I guess the functions in bdv-imageloader could have a way to accept a SharedQueue from whoever calls them. Or (less nice) there could be some global flag within bdv-imageloader to use a SharedQueue. I'd leave it up to you to come up with some good idea 😄  👍

Comment (KateMoreva) [reply src/main/java/de/embl/cba/mobie/n5/zarr/OMEZarrReader.java:43]: Ok. There will be no problems with accepting SharedQueue in loaders (they are called from MoBIE).

**Review by KateMoreva — APPROVED** [review]

**Pull request merged**